### DISCUSSION TITLE
How does the update process work, and how is data appended?

### DISCUSSION THREAD

**Discussion by FaFre** [osm2pgsql-dev/osm2pgsql#1795] [Q&A] [answered]
I use osm2pgsql together with the flex output. My Lua script processes and remixes the data with other data sources. After the insert I use Postgres triggers to further process geometries (I'm aware of the new features in 1.7, but the current function set is not enough for my scenario). 

At some point I want to append diff files to my database. How exactly does o2p handle the database updates? I could not find detailed information about that in the documentation (as most users don't have to care about this I guess). 

Following things are important to know, to keep the database in shape with my triggers:
- How does o2g actually know when data is outdated? Does it only rely on the diff files, or are also `SELECT`'s issued to the database to compare data?
- To update the data, o2p uses `UPDATE` or `DELETE`/`INSERT` statements?

It would be great to know about that, and also if the currently used update method is likely to change in the near future.

**Comment by joto** [comment, marked as answer]
The details are not documented on purpose, because they might change at any time. To find out what osm2pgsql is doing internally, you can use the `--log-sql` option though. :-)
